### PR TITLE
topology1: Add sample topology for i.MX8MP boards with wm8904 codec

### PR DESCRIPTION
**Pull request by a1batross**
Tested to work on Variscite Symphony-Board v1.3 with VAR-SOM-MX8M-PLUS

Sample device tree: https://gitlab.com/autogramma_multimedia_a/vendor/variscite/kernel_imx/-/commit/0e922e526bafe919fac9e89cf949e85c7b36b972

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by lgirdwood — COMMENTED** [review]
@dbaluta @paulstelian97 please review

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -212,6 +212,7 @@ set(TPLGS
 	"sof-imx8mp-wm8960\;sof-imx8mp-eq-iir-wm8960\;-DPPROC=eq-iir-volume"
 	"sof-imx8mp-wm8960\;sof-imx8mp-eq-fir-wm8960\;-DPPROC=eq-fir-volume"
 	"sof-imx8mp-wm8960\;sof-imx8mp-drc-wm8960\;-DPPROC=drc"
+	"sof-imx8mp-wm8904\;sof-imx8mp-wm8904\;-DPPROC=volume"
```
Comment (dbaluta) [inline tools/topology/topology1/CMakeLists.txt:216]: @a1batross thanks, this looks good! I'm glad to see other i.MX8mp platform supported. One question below

File: tools/topology/topology1/sof-imx8mp-wm8904.m4 [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# PCM Low Latency, id 0
+
+dnl PCM_DUPLEX_ADD(name, pcm_id, playback, capture)
+PCM_DUPLEX_ADD(Port0, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
+
+dnl DAI_CONFIG(type, idx, link_id, name, sai_config)
+DAI_CONFIG(SAI, 3, 0, sai3-wm8904-hifi,
+	SAI_CONFIG(I2S, SAI_CLOCK(mclk, 11565177, codec_mclk_in),
+		SAI_CLOCK(bclk, 1411200, codec_provider),
+		SAI_CLOCK(fsync, 44100, codec_provider),
```
Comment (dbaluta) [inline tools/topology/topology1/sof-imx8mp-wm8904.m4:80]: `11565177` how did you compute this? For now the clocks are managed from the Linux kernel side, so this has only a decorative meaning but I'm still curious.

also, do you have any other scenarios you plan to support with SOF?

Comment (a1batross) [reply tools/topology/topology1/sof-imx8mp-wm8904.m4:80]: @dbaluta 

Symphony Board uses just wm8904 by default, I modified the driver to make it more verbose and took all clocks from here. I'm not very familiar with wm8904 or SOF for now, so it's more or less rough values that however works.

Same clock value presented is hardcoded in DSP node in device tree.

Comment (a1batross) [reply tools/topology/topology1/sof-imx8mp-wm8904.m4:80]: @dbaluta we're planning to use SOF in our company in car multimedia project and as opensource advocate here I hope we're gonna upstream most significant changes. :)

Comment (dbaluta) [reply tools/topology/topology1/sof-imx8mp-wm8904.m4:80]: @a1batross All right, great! That was also my thinking when doing the patches for wm8960 :) so welcome to the club.

**Review by dbaluta — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by dbaluta** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@lgirdwood I think we can merge this not sure why CI is blocked!

**Pull request merged**